### PR TITLE
Get listing by id, get all bookings, create booking routes

### LINKED ISSUE #18: Create Booking, Get All Bookings, Get Listing by Listing Id Routes

### PR DESCRIPTION
**Pull request by cruiznunez**
- Added route that gets all bookings (returns array of booking objects)

- Added route that gets listing by a given listing id (takes listingId as a parameter, validates it, and returns a booking object)

- Added route that creates a booking where the request body contains startTimestamp, endTimestamp, bookerId, listingId, and numOfPeople such as:
{
    "startTimestamp" : "2022-12-25T06:42:15Z",
    "endTimestamp" : "2022-12-25T08:30:40Z",
    "bookerId" : "AqF6Rq4icHRufJdoqL9GbM5UFkF2",
    "listingId" : "52E7N5WDdIbjctOpiMaK",
    "numOfPeople" : 10
} 
where the startTimestamp and endTimestamp is validated to be a valid date object, bookerId to be a valid existing user in the user collection, listingId to be a valid existing listing id, and numOfPeople to be a valid integer. Additionally, more validation is done on the startTimestamp and endTimestamp to ensure they are valid date, both in the present or the future, endTimestamp is not before startTimestamp. This will also check to ensure there arent any conflicting bookings when creating a new booking.

- Added validation for listingId and numOfPeople

### CODE REVIEW

File: server/src/routes/bookings.ts [hunk 1]
```diff
@@ -34,3 +37,123 @@ bookingRoutes.post(
...
+
+  new xss.FilterXSS().process(req.body.numOfPeople).trim();
+  new xss.FilterXSS().process(req.body.endTimestamp).trim();
+  new xss.FilterXSS().process(req.body.startTimestamp).trim();
+
+  const bookerId = new xss.FilterXSS().process(req.body.bookerId).trim();
+  const listingId = new xss.FilterXSS().process(req.body.listingId).trim();
+  const numOfPeople = req.body.numOfPeople;
+  const endTimestamp = req.body.endTimestamp;
+  const startTimestamp = req.body.startTimestamp;
```
Comment (LDomadia) [inline server/src/routes/bookings.ts]: For "CreateBooking":

1. For `numOfPeople`, `endTimestamp`, and `startTimestamp`, the XSS filter is not applied to the constants, which means unsanitized inputs can still be added to the DB

2. This input caused an error "Dates to be book conflict", and the opposite passed. Maybe the validation is backwards?
```
"startTimestamp": "2022-12-25T06:42:15Z",
"endTimestamp": "2022-12-25T08:30:40Z"
```

File: server/src/routes/bookings.ts [hunk 2]
```diff
@@ -34,3 +37,123 @@ bookingRoutes.post(
...
+      });
+    });
+  } catch (e) {
+    res.status(409).json({ message : e});
+    return;
+  }
+  
+
+  const diffInDays = Math.round((endDateJustDate.getTime() - startDateJustDate.getTime()) / (1000 * 60 * 60 * 24));
+  const booking = await bookingsData.createBooking(bookerId, listingId, numOfPeople, listingFound.owner, (listingFound.price * (diffInDays + 1)), endDateJustDate.toString(), startDateJustDate.toString());
```
Comment (uityh) [inline server/src/routes/bookings.ts:157]: I think listingFound.owner needs to be listingFound.ownerId.

File: server/src/routes/bookings.ts [hunk 3]
```diff
@@ -34,3 +37,123 @@ bookingRoutes.post(
...
+
+  res.status(200).json(bookings);
+});
+
+bookingRoutes.post("/create", async (req: Request, res: Response) => {
+  console.log("POST /bookings/create");
+
+  new xss.FilterXSS().process(req.body.numOfPeople).trim();
+  new xss.FilterXSS().process(req.body.endTimestamp).trim();
+  new xss.FilterXSS().process(req.body.startTimestamp).trim();
```
Comment (uityh) [inline server/src/routes/bookings.ts]: Shouldn't the timestamps be set equal to the new xss stuff?

**Review by uityh — CHANGES_REQUESTED** [review]

File: server/src/routes/bookings.ts (hunk 2 above)
Comment (uityh) [inline server/src/routes/bookings.ts]: listingFound.owner should be listingFound.ownerId

**Review by uityh — CHANGES_REQUESTED** [review]

File: server/src/routes/bookings.ts (hunk 2 above)
Comment (cruiznunez) [reply server/src/routes/bookings.ts]: listing has an owner field not an ownerId field

File: server/src/routes/bookings.ts (hunk 3 above)
Comment (cruiznunez) [reply server/src/routes/bookings.ts]: made changes to go with this

**Review by LDomadia — APPROVED** [review]

**Review by uityh — APPROVED** [review]

**Pull request merged**